app/src/main/java/tech/jpco/qen/viewModel/QenViewModel.kt

```diff
@@ -0,0 +1,128 @@
+package tech.jpco.qen.viewModel
+
+import android.util.Log
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.ViewModel
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.PublishSubject
+import tech.jpco.qen.model.PagesRepository
+
+val Any.TAG
+    get() = this::class.simpleName ?: "Anon"
+
+fun Any.iLogger(output: String, obj: Any? = Unit) {
+    val name = Thread.currentThread().name
+    val objS = if (obj == Unit) "" else ": $obj"
+    Log.d(TAG, "$output$objS on ${name.substring(0, 1).toUpperCase()}${name.substring(1)}")
+}
+
+
+class QenViewModel : ViewModel() {
+    private val cd = CompositeDisposable()
+
+    //(Parallel) Dual Subject architecture per https://speakerdeck.com/oldergod/model-view-intent-for-android
+    /* In theory, it keeps an "alive" data stream from UI interaction to UI output, allowing an event in flight to
+     * be resubscribed by a recreated view before it has finished its journey.
+     */
+    //TODO test the above description!
+    private val touchesIn = PublishSubject.create<DrawPoint>()
+    private val metaIn = PublishSubject.create<MetaEvent>()
+    private val mTouchesOut = PublishSubject.create<DrawPoint>()
```

Comment:
Any suggestions for an alternative convention? Python-style `_`-prefixing?

Reply:
The simple convention of naming a property after its type, which doesn't always yield very descriptive names (e.g. `val int: Int`) could be used here to provide extra context. Since `touchesOut` is the public API of this class, it doesn't need to change, however, I would suggest changing `mTouchesOut` to `touchesOutSubject`. This provides an implicit link by name to the public property `touchesOut`, but also makes its internal usage more clear in a way that `mTouchesOut` does not.

Reply:
I like the conclusion, and I like the reasoning! A++, would consult again.